### PR TITLE
chore: add detailed documentation of custom monitoring [skip ci]

### LINKED ISSUE #153: [docs] Document CustomMonitoring in more detail
### Feature scope

docs

### Describe your suggested feature

Document custom monitoring in more details.

* provide example for each "mode" (metric, metric with alarm, metric search..)
* include anomaly detection

### PR DESCRIPTION
**Pull request by voho**
Fixes #153 

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -179,13 +179,101 @@ All the widgets will have the same size, which is chosen based on the number of
 Custom metric monitoring can be created for simple metrics, simple metrics with anomaly detection and search metrics.
 The first two also support alarming.
 
-### Custom monitoring
+#### Example: simple metrics
+
+```ts
+monitorCustom({
+    title: "Three simple metrics",
+    metrics: [m1, m2, m3]
```
Comment (echeung-amzn) [inline README.md:187]: It'd probably help to show a sample of a metric, or just add a comment stating what sort of metric classes/interfaces are allowed.

File: README.md [hunk 2]
```diff
@@ -179,13 +179,101 @@ All the widgets will have the same size, which is chosen based on the number of
...
+            }
+        }
+    ]
+})
+```
+
+#### Example: metric with anomaly detection
+
+In this case, only one metric is supported. 
+There cannot be multiple metrics rendered with anomaly detection in a single widget (CloudWatch limitation).
```
Comment (echeung-amzn) [inline README.md:214]:
```suggestion
Multiple metrics cannot be rendered with anomaly detection in a single widget due to a CloudWatch limitation.
```

File: README.md [hunk 3]
```diff
@@ -179,13 +179,101 @@ All the widgets will have the same size, which is chosen based on the number of
...
+
+#### Example: search metrics
+
+```ts
+monitorCustom({
+    title: "Metric search",
+    metrics: [
+        {
+            searchQuery: "My.Prefix.",
+            dimensions: {
```
Comment (echeung-amzn) [inline README.md:258]: Isn't it `dimensionsMap` now with CDK v2?

File: README.md [hunk 4]
```diff
@@ -179,13 +179,101 @@ All the widgets will have the same size, which is chosen based on the number of
...
+                FirstDimension: "FirstDimensionValue",
+                // allow any value for the given dimension
+                SecondDimension: undefined
```
Comment (echeung-amzn) [inline README.md:261]: This would have to be `undefined as unknown as string` to satisfy the type of `DimensionsMap`.

**Review by echeung-amzn — CHANGES_REQUESTED** [review]
It may be worth also just linking to the unit tests, which we could also potentially clean up a bit so it better serves as examples.

File: README.md (hunk 4 above)
Comment (voho) [reply README.md:261]: Wow. That seems bad. We should probably introduce some magical constant instead, like `AnyDimensionValue`.

File: README.md (hunk 1 above)
Comment (voho) [reply README.md:187]: I see, so you mean for m1, etc?

Comment (echeung-amzn) [reply README.md:187]: Yeah. A comment describing the allowed types should be sufficient.

**Review by echeung-amzn — APPROVED** [review]

**Pull request merged**